code/game/machinery/vending.dm

```diff
@@ -148,34 +147,31 @@
 
 			product_records.Add(product)
 
-/obj/machinery/vending/Destroy()
-	QDEL_NULL(wires)
-	QDEL_NULL(coin)
-	return ..()
-
-/obj/machinery/vending/ex_act(severity)
+/obj/machinery/vending/ex_act(severity) //TO-DO-OBJECT-DAMAGE: Kill off when everything is damageable

```

Comment:
I've left this message here for now.

Basically, since we have the object damaging system awkwardly put into place....and since `ex_act` damaging things is done at the `obj` level, but there's often different `ex_act`'s for machinery, we have to copy-paste what is normally at the object level here.

This is NOT ideal and is very gross, but barring someone going through and implementing the entire object damaging system all at once, this will suffice for now. 

If you understand how the system works and wish to do something similar, leave this exact same line so that when the system is "fully online" so to speak, they can go back and delete this crap.